### PR TITLE
fix: limit feedback message length

### LINKED ISSUE #4680: feedback: Restrict message length to 4096 characters
### Description

This will require a character counter underneath the text field, and new validation/error reporting logic

Need to include testing against our backend what happens with unicode precomposed characters: https://en.wikipedia.org/wiki/Precomposed_character#:~:text=A%20precomposed%20character%20(alternatively%20composite,letter%20e%20with%20acute%20accent).

### PR DESCRIPTION
**Pull request by llirik0**
## :scroll: Description

- Limit managed user-feedback messages to 4096 Unicode scalars, matching the backend's Python code-point validation.
- Show an accessible live character counter below the message field.
- Highlight the counter and present a specific validation error when the limit is exceeded.

## :bulb: Motivation and Context

The Sentry backend rejects feedback messages longer than 4096 code points. The iOS managed feedback form currently allows those messages to be submitted without warning, so feedback can be dropped after the user finishes the form.

Unicode-scalar counting intentionally matches backend `len(message)` behavior for precomposed and decomposed input.

Fixes #4680

## :green_heart: How did you test it?

```sh
./scripts/sentry-xcodebuild.sh \
  --platform iOS \
  --command test \
  --configuration Test \
  --ref fix/feedback-message-limit \
  --destination 'platform=iOS Simulator,id=50ED21A6-2279-4432-B0F0-09E8928CC1EE' \
  --only-testing 'SentryTests/SentryFeedbackTests'
```

Result: 38 tests passed with 0 failures on iOS 26.2. Coverage includes exact 4096/4097 Unicode-scalar boundaries for decomposed input, nil UIKit text handling, missing-required-field precedence, theme font customization, counter accessibility, and submit-error presentation. A sabotage run using Swift grapheme-cluster counting made the 4097-scalar decomposed regression test fail, then pass after restoring Unicode-scalar counting.

```sh
make format
make lint
make analyze
make build-ios FOR_AGENTS=true \
  IOS_SIMULATOR_OS=26.2 \
  IOS_DEVICE_NAME='Sentry OSS iPhone 17 Pro'
```

All commands passed. The destination overrides are required because this machine does not have the repository-default iOS 18.4 runtime.

## :pencil: Checklist

- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] I updated the wizard if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.
- [x] Public API changes reviewed by another Mobile SDK team member or implemented according to the develop docs spec.
- [x] If I added a new public API, I also added it to the SentryObjC wrapper.

Demo:

https://github.com/user-attachments/assets/567a10bc-429c-49bc-a9dd-311bd1e7f91b

### CODE REVIEW

**Comment by NinjaLikesCheez** [comment]
Overall this looks good - thank you for your contribution! I've kicked off the CI and there's a merge conflict to resolve, could you attach a screenshot of the UI so we can see what it looks like? I'll try get to testing it on device later today either so if I get there first I'll add a screenshot for others.

Thanks again!

**Comment by llirik0** [comment]
> Overall this looks good - thank you for your contribution! I've kicked off the CI and there's a merge conflict to resolve, could you attach a screenshot of the UI so we can see what it looks like? I'll try get to testing it on device later today either so if I get there first I'll add a screenshot for others.
> 
> Thanks again!

Thanks for the review! I’ve resolved the merge conflict and attached a short demo below showing the character counter and validation state. Please let me know if anything else needed here!

**Review by NinjaLikesCheez — APPROVED** [review]
LGTM - thanks!

File: Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackFormViewModel.swift [hunk 1]
```diff
@@ -399,6 +414,17 @@ extension SentryUserFeedbackFormViewModel {
         case .failure(let error): submitButton.accessibilityHint = error.errorDescription
         }
     }
+
+    func updateMessageCharacterCount() {
+        updateMessageCharacterCount(label: messageCharacterCountLabel)
+    }
+
+    private func updateMessageCharacterCount(label: UILabel) {
```
Comment (philprime) [inline Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackFormViewModel.swift:422]: `h`: Our user feedback UI is standardized across SDKs, so we must do an internal alignment with other SDK maintainers if adding this footer caption label is fine.

**Review by philprime — COMMENTED** [review]
Thank you @llirik0 for opening up this issue. I see the problem and how your implementation can help with sending valid feedback only. 

As this is PR is also extending the standard user feedback UI with a character count label, I will need to raise this internally to gather feedback

File: Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackFormViewModel.swift (hunk 1 above)
Comment (llirik0) [reply Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackFormViewModel.swift:422]: sg, please let me know if anything needs to be changed. ty

**Comment by philprime** [comment]
Created https://github.com/getsentry/sentry-docs/pull/19424 as the matter of discussion on how to align cross-SDK behavior for the client-side feedback input validation. I'll let you know when we have a decision.

**Comment by philprime** [comment]
Thanks again for the work so far. We've now aligned internally on the specification and documented them here:

https://develop.sentry.dev/sdk/telemetry/feedbacks/#client-side-input-validation

Please take a look and update the pull request to match what is defined there.

**Comment by llirik0** [comment]
> Thanks again for the work so far. We've now aligned internally on the specification and documented them here:
> 
> https://develop.sentry.dev/sdk/telemetry/feedbacks/#client-side-input-validation
> 
> Please take a look and update the pull request to match what is defined there.

thanks for the update! I reviewed the finalized spec and updated the pr accordingly. It now validates the final payload after event processing, rejects whitespace only or over limit messages and invalid associated event id's, reports discarded feedback, and aligns the form’s counter, accessibility, correction, and IME behavior with the spec.

rebased it onto the latest main, added regression coverage, and reran the relevant tests and quality checks. Everything is green. Please take another look when you have a chance!
